editor/plugins/script_editor_plugin.cpp

```diff
@@ -4160,10 +4216,63 @@ void ScriptEditorPlugin::apply_changes() {
 
 void ScriptEditorPlugin::set_window_layout(Ref<ConfigFile> p_layout) {
 	script_editor->set_window_layout(p_layout);
+
+	if (EDITOR_GET("interface/multi_window/restore_windows_on_load") && window_wrapper->is_window_available() && p_layout->has_section_key("ScriptEditor", "window_rect")) {
+		// Restore the detached window.
+		Rect2i window_rect = p_layout->get_value("ScriptEditor", "window_rect", Rect2i());
+		int screen = p_layout->get_value("ScriptEditor", "window_screen", -1);
+		Rect2i screen_rect = p_layout->get_value("ScriptEditor", "window_screen_rect", Rect2i());
```

Comment:
This code seems to repeat in 3 different places. Maybe it could be a method of WindowWrapper and the 3 settings could be its arguments.